Zend/zend_vm_def.h

```diff
@@ -5583,6 +5583,111 @@ ZEND_VM_C_LABEL(num_index):
 	ZEND_VM_NEXT_OPCODE_CHECK_EXCEPTION();
 }
 
+ZEND_VM_HANDLER(208, ZEND_ADD_ARRAY_UNPACK, ANY, ANY)
+{
+	USE_OPLINE
+	zend_free_op free_op1;
+	zval *op1, new_op1;
+	
+	SAVE_OPLINE();
+	op1 = GET_OP1_ZVAL_PTR_UNDEF(BP_VAR_R);
+	
+ZEND_VM_C_LABEL(add_again):
+	if (EXPECTED(Z_TYPE_P(op1) == IS_ARRAY)) {
+		HashTable *ht = Z_ARRVAL_P(op1);
+		zval *val;
+		zend_string *key;
+
+		ZEND_HASH_FOREACH_STR_KEY_VAL(ht, key, val) {
+			Z_TRY_ADDREF_P(val);
+
+			if (key) {
+				zend_throw_error(NULL, "Cannot unpack array with string keys");
+				FREE_OP1();
+				HANDLE_EXCEPTION();
+			} else {
+				if (!zend_hash_next_index_insert(Z_ARRVAL_P(EX_VAR(opline->result.var)), val)) {
```

Comment:
I don't think this code-path has test coverage right now.

Reply:
I'm also wondering if we should stop after the first insertion fails, as all following will also fail.

Reply:
`zend_cannot_add_element` only throws warning, which won't stop running usually

Reply:
Right, but does that make sense? I don't think it's particularly useful to throw this warning N times, just once should be enough. (E.g. we only throw it once in array_fill.)